server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalGeoCentroidTests.java

```diff
@@ -67,6 +70,16 @@ protected void assertReduced(InternalGeoCentroid reduced, List<InternalGeoCentro
         assertEquals(totalCount, reduced.count());
     }
 
+    public void testReduceMaxCount() {
+        InternalGeoCentroid maxValueGeoCentroid = new InternalGeoCentroid("agg", new GeoPoint(10, 0),
+            Long.MAX_VALUE, Collections.emptyMap());
+        InternalGeoCentroid reducingGeoCentroid = new InternalGeoCentroid("agg", new GeoPoint(10, 0),
+            randomNonNegativeLong(), Collections.emptyMap());
+        InternalGeoCentroid reducedGeoCentroid = reducingGeoCentroid
```

Comment:
Usually the reducing result is in the list, right?

Reply:
I suppose I was not really interested in the semantics of how this method was used in the real world, and more interested in the reduce logic itself -- as it is returned. should I change this?

Reply:
It'd make me feel better if you did `maxValueGeoCentroid.reduce(singletonList(maxValueGeoCentroid), null)`.

Reply:
done